test/metabase/lib/fe_util_test.cljc

```diff
@@ -121,6 +123,216 @@
+          (is (=? parts (lib.fe-util/coordinate-filter-parts query -1 (lib.fe-util/coordinate-filter-clause operator
+                                                                                                            column
+                                                                                                            longitude-column
+                                                                                                            values)))))))
+    (testing "unsupported clauses"
+      (are [clause] (= nil (lib.fe-util/coordinate-filter-parts query -1 clause))
+        (lib.filter/= 10 lat-column)
+        (lib.filter/is-null lat-column)
+        (lib.filter/= (meta/field-metadata :orders :total) 10)
+        (lib.expression/+ lat-column 10)))))
+
+(deftest ^:parallel boolean-filter-parts-test
+  (let [query  (-> lib.tu/venues-query
+                   (lib.expression/expression "Boolean"
+                                              (lib.filter/is-empty (meta/field-metadata :venues :name))))
+        column (m/find-first #(= (:name %) "Boolean") (lib.filter/filterable-columns query))]
+    (testing "clause to parts rountrip"
+      (doseq [[clause parts] {(lib.filter/is-null column)       {:operator :is-null, :column column, :values []}
+                              (lib.filter/not-null column)      {:operator :not-null, :column column, :values []}
+                              (lib.filter/= column true)        {:operator :=, :column column, :values [true]}
+                              (lib.filter/= column false)       {:operator :=, :column column, :values [false]}}]
+        (let [{:keys [operator column values]} parts
+              expected {:operator operator, :column (select-keys column [:name]), :values values}]
+          (is (=? expected (lib.fe-util/boolean-filter-parts query -1 clause)))
+          (is (=? expected (lib.fe-util/boolean-filter-parts query -1 (lib.fe-util/boolean-filter-clause operator
+                                                                                                         column
+                                                                                                         values)))))))
+    (testing "unsupported clauses"
+      (are [clause] (= nil (lib.fe-util/boolean-filter-parts query -1 clause))
+        (lib.filter/= true column)
+        (lib.filter/!= column true)
+        (lib.filter/is-null (meta/field-metadata :venues :name))))))
+
+(deftest ^:parallel relative-date-filter-parts-test
+  (let [query lib.tu/venues-query
+        column (meta/field-metadata :checkins :date)]
+    (testing "clause to parts rountrip"
+      (doseq [[clause parts] {(lib.filter/time-interval column :current :day)
+                              {:column column
+                               :value :current
+                               :unit :day
+                               :offset-value nil
+                               :offset-unit nil
```

Comment:
If the `nil` values are dropped from `relative-date-filter-parts`, then their absence can still be checked in a `=?` clause with `:offset-value (symbol "nil #_\"key is not present\"")` on the LHS.

Reply:
Fixed